lib/cts/cases/reindex.rb

```diff
@@ -11,8 +11,26 @@ def initialize(logger, options = {})
 
     def call
       logger.info Benchmark::CAPTION
-      logger.info Benchmark.measure { @count = Case::Base.update_all_indexes.count }
+      logger.info Benchmark.measure { @count = process_indexes }
       logger.info "\n#{@count} cases reindexed"
     end
+
+    private 
+
+    def process_indexes
+      if @options[:non_indexed]
+        size = @options.fetch(:size)
+        logger.info "Start to query the cases with size #{size}."
+        cases = Case::Base.where("document_tsvector is NULL").limit(size)
```

Comment:
So the idea is to pass the `reindex` task two options or none, is that right? Just wondering what happens if the `--size` option is omitted or you pass a string to it?

Should the method exit if you don't pass the size to it?

Reply:
Interesting contrast between `@options[:non_indexed]` and `@options.fetch(:size)` - wondering from interest why use both syntaxes? They seem to behave slightly differently in edge cases e.g.  https://gist.github.com/stevenharman/5664318

Reply:
hmm,  didn't think about it too much, somehow I am sort of thinking the "fetch" similar as the "get" in python dict operator which will return nil even the key is not included, and [key] will raise a exception if key is not included, but after I checked the link, it is not what I thought,  and for this case,  2 doesn't make different, in that case let me change to [key]